daemon/fqdn.go

```diff
@@ -100,6 +101,17 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState) (err err
 		fqdn.StartDNSPoller(d.dnsPoller)
 	}
 
+	// Controller to cleanup TTL expired entries from the DNS policies.
+	controller.NewManager().UpdateController("dns-ttl-cleanup-job", controller.ControllerParams{
```

Comment:
You were right: putting this here makes things a little complicated. We now need a controller for the Endpoint.DNSHistory caches(unless some of the changes below around replacing cleanup entries go in, then it might end up with a steady memory usage).